padertorch/train/hooks.py

```diff
@@ -445,22 +449,18 @@ def finalize_summary(self, trainer):
     def pre_step(self, trainer: 'pt.Trainer'):
         if self.trigger(iteration=trainer.iteration, epoch=trainer.epoch):
             self.run_validation(trainer)
+            self.last_validation = trainer.iteration
+        if (
+            self.early_stopping_patience is not None
+            and self.n_degradations > self.early_stopping_patience
+        ):
+            print(f'Early stopping after {trainer.epoch} epochs and'
+                  f' {trainer.iteration} iterations')
+            raise StopTraining
 
     def run_validation(self, trainer: 'pt.Trainer'):
         ckpt_dir = trainer.checkpoint_dir
-        if self._json_file is None:
-            self._json_file = ckpt_dir / self._json_filename
-            if self._json_file.exists():
-                self.load_validation_state()
         ckpt_path: Path = trainer.default_checkpoint_path()
```

Comment:
Can you add a comment that `ckpt_path` does not exist at this moment, because it will be written later with the state of this hook?

Reply:
Done